hibernate-core/src/main/java/org/hibernate/loader/ColumnEntityAliases.java

```diff
@@ -33,8 +33,12 @@ public ColumnEntityAliases(
 	protected String getDiscriminatorAlias(Loadable persister, String suffix) {
 		return persister.getDiscriminatorColumnName();
 	}
-	
+
 	protected String[] getPropertyAliases(Loadable persister, int j) {
-		return persister.getPropertyColumnNames(j);
+		String[] propertyColumnNames = persister.getPropertyColumnNames(j);
+		if (propertyColumnNames!=null && propertyColumnNames.length==1 && propertyColumnNames[0]==null) {
```

Comment:
It would be great to use space chars consistently with existing coding format as
`
if ( propertyColumnNames != null && propertyColumnNames.length == 1 && propertyColumnNames[0] == null )
`

Reply:
Not sure whether using `Arrays.equals()` might improve the code readability as
`
if ( Arrays.equals( propertyColumnNames, new String[] { null } ) {
    ... ...
}
`

Reply:
Inconsistent spaces fixed and switched to Arrays.equals as suggested.